public/sections/todo/directives/todo.create.modal.directive.js

```diff
@@ -0,0 +1,67 @@
+angular.module('todoApp')
+    .directive('myTodoCreateModal', myTodoCreateModal)
+    .directive('myTodoCreateModalBtn', myTodoCreateModalBtn);
+
+function myTodoCreateModal(){
+    return {
+        scope : {
+            todoModalId : '@',
+            onSuccess : '&'
+        },
+        templateUrl : 'sections/todo/directives/todo.create.modal.html',
+        controller : ['$scope', 'Todo', myTodoCreateModalController]
+    }
+}
+
+function myTodoCreateModalBtn(){
+    return {
+        scope : {
+            todoModalId : '@',
+            classes : '@?'
+        },
+        transclude : true,
+        templateUrl : 'sections/todo/directives/todo.create.modal.btn.html',
+        controller : ['$scope', myTodoCreateModalBtnController]
+    }
+}
+
+function myTodoCreateModalBtnController($scope){
+    $scope.openModal = openModal;
+
+    function openModal(){
+        $('#' + $scope.todoModalId).openModal();
+    }
+}
+
+function myTodoCreateModalController($scope, Todo){
+    angular.element(document).ready(function () {
+        initMaterializeDesign()
+    });
+
+    $scope.create = create;
+
+    $scope.todo = {
+        name : ''
+    };
+
+    function refreshTodo(){
+        $scope.todo.name = '';
+    }
+
+    function create(){
+        console.log($scope.todo)
+        Todo.save($scope.todo, function(data){
+            console.log(data);
+            $scope.onSuccess(data);
+            refreshTodo();
+        }, function(err){
+            console.log(err);
```

Comment:
Hmmm, we should probably handle errors more gracefully, taking the error and notifying the user somehow that something went wrong so they know what happened and why.